pkg/controllers/user/nsserviceaccount/nssvcaccnt.go

```diff
@@ -0,0 +1,85 @@
+package nsserviceaccount
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type namespaceSvcAccountHandler struct {
+	clusterCtx *config.UserContext
+}
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	logrus.Debugf("Registering namespaceHandler for checking default serviceaccount")
+	nsh := &namespaceSvcAccountHandler{
+		clusterCtx: cluster,
+	}
+	cluster.Core.Namespaces("").AddHandler(ctx, "namespaceHandler", nsh.Sync)
+}
+
+func (nsh *namespaceSvcAccountHandler) Sync(key string, ns *corev1.Namespace) (runtime.Object, error) {
+	if ns == nil || ns.DeletionTimestamp != nil {
+		return nil, nil
+	}
+	logrus.Debugf("namespaceSvcAccountHandler: Sync: key=%v, ns=%+v", key, *ns)
+
+	//handle default svcAccount of system namespaces
+	if err := nsh.handleSystemNS(key); err != nil {
+		logrus.Errorf("namespaceSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=%v, err=%v", key, err)
+	}
+	return nil, nil
+}
+
+func (nsh *namespaceSvcAccountHandler) handleSystemNS(namespace string) error {
+
+	if namespace == "kube-system" || namespace == "default" || !nsh.isSystemNS(namespace) {
+		return nil
+	}
+
+	var svcAccounts clientv1.ServiceAccountInterface
+	svcAccounts = nsh.clusterCtx.K8sClient.CoreV1().ServiceAccounts(namespace)
+
+	defSvcAccnt, err := svcAccounts.Get("default", metav1.GetOptions{})
+	if err != nil {
+		logrus.Errorf("namespaceSvcAccountHandler: error listing serviceaccount flag: Sync: key=%v, err=%+v", namespace, err)
+		return err
+	}
+
+	if defSvcAccnt.AutomountServiceAccountToken != nil && *defSvcAccnt.AutomountServiceAccountToken == false {
+		return nil
+	}
+
+	automountServiceAccountToken := false
+	defSvcAccnt.AutomountServiceAccountToken = &automountServiceAccountToken
+	defSvcAccnt.Namespace = namespace
+	logrus.Debugf("namespaceSvcAccountHandler: updating default serviceaccount key=%v", defSvcAccnt)
```

Comment:
Perhaps this should be logrus.info ? If they setup a serviceAccount and then we change it they could get confused or it could break something. Although they shouldn't be doing that in system ns. But at least if it's in the logs they would get a warning. Up for debate. 

Also might want to specify what changed, ie:
> setting automount to false on default serviceaccount...

Reply:
@cbron We are only changing "default" serviceaccounts or rancher system namespaces. So I think logrus.debugf should be fine since we are not changing any user namespace or serviceaccount?

Reply:
Yea, its very unlikely to affect a customer. If they do mess with it they should know we are prone to change our own system.